packages/client/lib/rpc/modules/engine.ts

```diff
@@ -277,17 +346,15 @@ export class Engine {
    *   3. validationError: String|null - validation error message
    */
   async newPayloadV1(params: [ExecutionPayloadV1]): Promise<PayloadStatusV1> {
-    const [payloadData] = params
-    const {
-      blockNumber: number,
-      receiptsRoot: receiptTrie,
-      prevRandao: mixHash,
-      feeRecipient: coinbase,
-      transactions,
-      parentHash,
-      blockHash,
-    } = payloadData
-    const { chainCommon: common } = this.config
+    const [payload] = params
+    const { parentHash, blockHash } = payload
+
+    const block = await assembleBlock(payload, this.chain)
+    if (!(block instanceof Block)) {
+      // Return error response
+      this.connectionManager.lastNewPayload({ payload, response: block })
+      return block
```

Comment:
What do you think about making the assemble block to return
```typescript
{
  block: Block,
  error: PayloadStatusV1
}
```
and then we can do something like
```typescript
const { block, error: assembleBlockError } = await assembleBlock(payload, this.chain)
if (assembleBlockError) {
  // Return error response
  ...
}
```
The reason for this is that since we are not returning a block but instead, an error object; (IMO) this would make it easier to read?

Reply:
Ah good idea, I will update that :)

Reply:
how's this https://github.com/ethereumjs/ethereumjs-monorepo/pull/1830/commits/6355cc72fe90e5d9b3438b09c3cf72e624b6a77d